src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -172,6 +182,7 @@ def get_custom_sdk(custom_module, client_factory):
         g.show_command('show', 'get')
         g.custom_command('list', 'list_workspaces')
         g.custom_command('create', 'create_workspace', supports_no_wait=True)
+        g.custom_command('activate', 'activate_workspace', client_factory=cf_synapse_client_cmk_factory, supports_no_wait=True)
         g.custom_command('update', 'update_workspace', supports_no_wait=True)
```

Comment:
For what purpose that you make it a separate command rather than integrating it into `az synapse workspace update`?

Reply:
az synapse workspace update using Workspaces_Update SDK
az synapse workspace activate using Workspaces_CreateOrUpdate SDK

https://github.com/Azure/azure-cli/pull/16224/files/869129e44b85c3dbeb8c49ed7501b8cc7308703f#diff-51cec1d07d29bd22f589a3292d710c02d685386db30e10c283a41fffd6f562d6R94

Reply:
I think there must be some reasons that you use different SDK update method. Could you please share more details?

Reply:
Sorry didn't descript it clearly.
activate_workspace using the SDK defined at Key.json(CreateOrUpdate)
```
g.custom_command('activate', 'activate_workspace', client_factory=cf_synapse_client_cmk_factory, supports_no_wait=True)
```
update_workspace using the SDK defined at workspace.json(Update)

That means these two cmdlets will use different client to send request. We can put activate_workspace 's logic into `az synapse workspace key update`. But I don't think it may be more clear. How do you design at Powershll?

Reply:
Will move it to `az synapse workspace key update` according to offline discussion.